src/ansiblelint/rules/name.py

```diff
@@ -74,6 +74,19 @@ def _check_name(
         # lowercase letter, so we ignore anything else. On Unicode isupper()
         # is not necessarily the opposite of islower()
         results = []
+        task_components = name.split("|")
+        if len(task_components) > 1:
+            prefix_name = task_components[0].strip()
+            if prefix_name == "main":
```

Comment:
You also need to check if the prefix is matching allowed pattern. I would suggest to take a different approach and build the expected prefix based on prefix template string. If it does not match we can raise an error with tag `name[prefix]` and a clear message.

This also means that you can easily check if filename base name (w/o extension) is main and make the expected prefix = "".

You will be able to generate an error message like "Task name should use `<foo>` prefix in order to make it easy to identify.